### PR TITLE
MB 27 Copied over boom button and implemented timed deletion

### PR DESCRIPTION
**Pull request by MattKirchh**
Boom button will be destroyed 3 seconds after it is placed.

### CODE REVIEW

File: src/main/java/com/mineboundteam/minebound/block/BoomButton.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                // Block will be destroyed 3 seconds after being placed.
+                for (int i = 3; i > 0; i--) {
+                    Thread.sleep(1000);
+                }
+                level.destroyBlock(blockPos, false);
+            } catch (InterruptedException interruptedException) {
+                System.err.println(interruptedException.getMessage());
+            }
+        }).start();
+    }
```
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/block/BoomButton.java:34]: This is the important part of this PR.

Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/block/BoomButton.java:34]: This is the core part of this PR.

**Comment by insertStringHere** [comment]
Is there a way this can be done with something like a scheduled tick event? I'm not sure we want to be making threads for every time we want to do this

File: src/main/java/com/mineboundteam/minebound/block/BoomButton.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public void onPlace(BlockState blockState, Level level, BlockPos blockPos, BlockState p_60569_, boolean p_60570_) {
+        // Schedule a tick for 60 ticks in the future (3 seconds, 20 ticks per second)
+        level.scheduleTick(blockPos, blockState.getBlock(), 60);
+    }
+
+    @Override
+    public void tick(BlockState p_60462_, ServerLevel level, BlockPos blockPos, Random p_60465_) {
+        // Runs when scheduled in onPlace and destroys the block
+        level.destroyBlock(blockPos, false);
+    }
```
Comment (MattKirchh) [inline src/main/java/com/mineboundteam/minebound/block/BoomButton.java:34]: Took forever to figure out, but it's been changed to schedule a tick for destroying the block. Here is the new code.

**Review by insertStringHere — APPROVED** [review]

**Pull request merged**